File: backend/app/schemas/wishlist.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, Field

from app.schemas.card import CardResponse


class WishlistAddRequest(BaseModel):
    pokemon_tcg_id: str
    target_price: Optional[float] = Field(default=None, ge=0)


class WishlistItemResponse(BaseModel):
    id: UUID
    card: CardResponse
    target_price: Optional[float] = None
    current_price: Optional[float] = None
    price_source: Optional[str] = None
    at_target: bool = False
    added_at: datetime

    model_config = {"from_attributes": True}


class WishlistResponse(BaseModel):
    items: list[WishlistItemResponse]

File: frontend/types/wishlist.ts
import { Card } from "./card";

export interface WishlistItem {
  id: string;
  card: Card;
  target_price?: number | null;
  current_price?: number | null;
  price_source?: string | null;
  at_target: boolean;
  added_at: string;
}

export interface WishlistResponse {
  items: WishlistItem[];
}
